frontend/src/_ui/Avatar/index.jsx

```diff
@@ -1,13 +1,30 @@
 import React from 'react';
+import { userService } from '@/_services';
+
+const Avatar = ({ text, image, avatarId, title = '', borderColor = '', borderShape }) => {
+  const [avatar, setAvatar] = React.useState();
+
+  React.useEffect(() => {
+    async function fetchAvatar() {
+      const blob = await userService.getAvatar(avatarId);
+      setAvatar(URL.createObjectURL(blob));
+    }
+    if (avatarId) fetchAvatar();
+
+    () => avatar && URL.revokeObjectURL(avatar);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [avatarId]);
 
-const Avatar = ({ text, image, title = '', borderColor = '' }) => {
   return (
     <span
       data-tip={title}
-      style={{ border: `1.5px solid ${borderColor}`, ...(image ? { backgroundImage: `url(${image})` } : {}) }}
-      className="avatar avatar-sm avatar-rounded animation-fade"
+      style={{
+        border: borderColor ? `1.5px solid ${borderColor}` : 'none',
+        ...(image || avatar ? { backgroundImage: `url(${avatar ?? image})` } : {}),
+      }}
+      className={`avatar avatar-sm ${borderShape === 'rounded' ? 'avatar-rounded' : ''} animation-fade`}
```

Comment:
Let's simplify this using cx package
```suggestion
      import cx from 'classnames'
      ...
      className={cx('vatar avatar-sm animation-fade', { 
        'avatar-rounded': borderShape === 'rounded',
     })}
```